web/src/components/core/FileViewer.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from "react";
+import { Popup } from "~/components/core";
+import { Alert, Bullseye, Spinner } from "@patternfly/react-core";
+import cockpit from "../../lib/cockpit";
+
+import "./fileviewer.scss";
+
+// file loading indicator
+const spinner = () => {
+  return (
+    <Bullseye>
+      <Spinner size="xl" />
+    </Bullseye>
+  );
+};
+
+export default function FileViewer({ file, title, onCloseCallback }) {
+  // the popup is visible
+  const [isOpen, setIsOpen] = useState(true);
+  // error message for failed load
+  const [error, setError] = useState(null);
+  // the file content
+  const [content, setContent] = useState("");
+  // current state
+  const [state, setState] = useState("init");
+
+  useEffect(() => {
+    if (state !== "init") return;
+
+    setState("loading");
+    // NOTE: reading non-existing files in cockpit does not fail, the result is null
+    // see https://cockpit-project.org/guide/latest/cockpit-file
+    cockpit.file(file).read()
+      .then((data) => {
+        setState("ready");
+        setContent(data);
+      })
+      .catch((data) => {
+        setState("ready");
+        setError(data.message);
+      });
+  }, [file, state]);
+
+  const close = () => {
+    setIsOpen(false);
+    if (onCloseCallback) onCloseCallback();
+  };
+
+  return (
+    <>
+      <Popup
+        isOpen={isOpen}
+        title={title || file}
+        variant="large"
```

Comment:
Not related with this PR, but leaving here as a note for myself. We have a `height` Popup prop that somehow collides with the variant. The intention of this prop is to limit the height and to avoid _"blinking"_ effect we have here when changing from loading to the file content. But the actual size for _height="large"_ looks not enough. We can address and improve the Popup behavior later.